lib/validate.js

```diff
@@ -27,13 +49,14 @@ module.exports = {
 		// Expose entries - must be done before requiring the webpack configuration
     const entries = {};
     const functions = this.serverless.service.getAllFunctions();
+    const allFilesInServicePath = listFiles(this.serverless.config.servicePath);
```

Comment:
Is it necessary to browse the whole service dir file tree here? I think that is very expensive - the directory might include node_modules etc.

Assuming that the handlers we have already point to the handler files we check, we know exactly where the files have to be. So just searching in getEntryExtension(handlerFile) for `handlerFile.*` should retrieve the full file automatically (and even shows us if it is ambiguous, which is an error on the user side if multiple of our allowed extensions are found). We do not need to be aware of the complete file hierarchy here.